File: halflife.js
function compute_half_life(NT, NO, T) {
  if (NT <= 0 || NO <= 0 || T <= 0 || NT >= NO) {
    return null;
  }
  const half_life = (T * Math.log(2)) / Math.log(NO / NT);
  return +half_life.toFixed(4);
}

File: halflife.py
import math

def compute_half_life(NT, NO, T):
    if not all(isinstance(i, int) and i > 0 for i in [NT, NO, T]) or NO <= NT:
        return None
    
    k = math.log(2) / T
    t_half = math.log(NO / NT) / k
    
    return round(t_half, 4)
